hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -92,7 +92,7 @@ public class ExternalSpillableMap<T extends Serializable, R> implements Map<T, R
   public ExternalSpillableMap(long maxInMemorySizeInBytes, String baseFilePath, SizeEstimator<T> keySizeEstimator,
                               SizeEstimator<R> valueSizeEstimator, DiskMapType diskMapType, CustomSerializer<R> valueSerializer,
                               boolean isCompressionEnabled, String loggingContext) throws IOException {
-    this.inMemoryMap = new HashMap<>();
+    this.inMemoryMap = new ConcurrentHashMap<>();
```

Comment:
Is it because the map is concurrently modified by multiple threads?  The class is marked as not thread-safe though.

Reply:
Yes, see the above response as well for why it needs to be threadsafe